src/zenml/integrations/kserve/model_deployers/kserve_model_deployer.py

```diff
@@ -0,0 +1,464 @@
+                        pass
+
+        if service:
+            # update an equivalent service in place
+            service.update(config)
+            logger.info(
+                f"Updating an existing KServe deployment service: {service}"
+            )
+        else:
+            # create a new service
+            service = KServeDeploymentService(config=config)
+            logger.info(f"Creating a new KServe deployment service: {service}")
+
+        # start the service which in turn provisions the KServe
+        # deployment server and waits for it to reach a ready state
+        service.start(timeout=timeout)
+        return service
+
+    def get_kserve_deployments(
+        self, labels: Dict[str, str]
+    ) -> List[V1beta1InferenceService]:
+        """Get a list of KServe deployments that match the supplied labels.
+
+        Args:
+            labels: a dictionary of labels to match against KServe deployments.
+
+        Returns:
+            A list of KServe deployments that match the supplied labels.
+
+        Raises:
+            RuntimeError: if an operational failure is encountered while
+        """
+        labels = labels or {}
+        label_selector = (
+            ",".join(f"{k}={v}" for k, v in labels.items()) if labels else None
+        )
+
+        namespace = (
+            self.kubernetes_namespace or utils.get_default_target_namespace()
+        )
+
+        try:
+            response = (
+                self.kserve_client.api_instance.list_namespaced_custom_object(
+                    constants.KSERVE_GROUP,
+                    constants.KSERVE_V1BETA1_VERSION,
+                    namespace,
+                    constants.KSERVE_PLURAL,
+                    label_selector=label_selector,
+                )
+            )
+        except client.rest.ApiException as e:
+            raise RuntimeError(
+                "Exception when retrieving KServe inference services\
+                %s\n"
+                % e
+            )
+
+        # TODO[CRITICAL]: de-serialize each item into a complete
+        #   V1beta1InferenceService object recursively using the OpenApi
+        #   schema (this doesn't work right now)
+        inference_services: List[V1beta1InferenceService] = []
+        for item in response.get("items", []):
+            snake_case_item = self._camel_to_snake(item)
+            inference_service = V1beta1InferenceService(**snake_case_item)
+            inference_services.append(inference_service)
+        return inference_services
+
+    def _camel_to_snake(self, obj: Dict[str, Any]) -> Dict[str, Any]:
+        if isinstance(obj, (str, int, float)):
+            return obj
+        if isinstance(obj, dict):
+            assert obj is not None
+            new = obj.__class__()
+            for k, v in obj.items():
+                new[self._convert_to_snake(k)] = self._camel_to_snake(v)
+        elif isinstance(obj, (list, set, tuple)):
+            assert obj is not None
```

Comment:
```suggestion
```

Reply:
@schustmi how can I ignore mypy error if I don't want to add `assert obj is not None`

Reply:
What error is mypy raising? The if condition above should already ensure that `obj` is not None since None is not an instance of anything.